Source/Modules/fortran.cxx

```diff
@@ -491,7 +491,25 @@ int FORTRAN::functionWrapper(Node *n)
         // Get the C type
         String* tm = get_attached_typemap(p, "ctype",
                                           WARN_FORTRAN_TYPEMAP_CTYPE_UNDEF);
-        Printv(f->def, prepend_comma, tm, " ", arg, NULL);
+        bool is_function = (strstr(Char(tm), " (*)(") != NULL);
+        if (!is_function)
+            Printv(f->def, prepend_comma, tm, " ", arg, NULL);
+        else
+        {
+            String* tm_mod = Copy(tm);
+
+            char* dest = (char*) malloc(Len(tm_mod) + 10);
+            dest[0] = 0;
+
+            strcat(dest, " (*");
+            strcat(dest, Char(arg));
+            strcat(dest, ")(");
+
+            Replace(tm_mod, " (*)(", dest, DOH_REPLACE_FIRST);
+            Printv(f->def, prepend_comma, tm_mod, NULL);
+            free(dest);
+            Delete(tm_mod);
+        }
```

Comment:
Instead of injecting C-library string manipulation into the code (which is basically not done anywhere else in the SWIG source), try to use their wacky string manipulation for consistency. Also, instead of grepping for a string to see if there's a function, use the SWIG-provided typemap matching functions.